base/applications/regedit/framewnd.c

```diff
@@ -164,10 +164,36 @@ void SetupStatusBar(HWND hWnd, BOOL bResize)
 
 void UpdateStatusBar(void)
 {
-    HKEY hKeyRoot;
-    LPCWSTR pszKeyPath = GetItemPath(g_pChildWnd->hTreeWnd, 0, &hKeyRoot);
+    HKEY hKeyRoot = NULL;
+    LPCWSTR pszKeyPath = NULL;
+    LPWSTR pszFullPath = NULL;
+    DWORD dwCbFullPath;
+
+    pszKeyPath = GetItemPath(g_pChildWnd->hTreeWnd, 0, &hKeyRoot);
+
+    LPCWSTR pszRootName = get_root_key_name(hKeyRoot);
 
-    SendMessageW(hStatusBar, SB_SETTEXTW, 0, (LPARAM)pszKeyPath);
+    dwCbFullPath = (wcslen(pszRootName) + 1 + wcslen(pszKeyPath) + 1) * sizeof(WCHAR);
+    pszFullPath = malloc(dwCbFullPath);
+
+    if (pszFullPath)
+    {
+        if (pszKeyPath[0] != UNICODE_NULL)
+        {
+            StringCbPrintfW(pszFullPath, dwCbFullPath, L"%s%s%s", pszRootName,
+                            ((pszKeyPath[0] == L'\\') ? L"" : L"\\"), pszKeyPath);
+        }
+        else
+        {
+            StringCbCopyW(pszFullPath, dwCbFullPath, pszRootName);
+        }
+    }
+
+    if (pszFullPath)
+    {
+        SendMessageW(hStatusBar, SB_SETTEXTW, 0, (LPARAM)pszFullPath);
+        free(pszFullPath);
+    }
```

Comment:
Checking whether the allocation succeeded and if not, return early. The rest of the code can then follow, without having to re-verify whether pszFullPath is not NULL, and this allows decreasing the level of indentation.
```suggestion
    if (!pszFullPath)
        return;

    if (pszKeyPath[0] != UNICODE_NULL)
    {
        StringCbPrintfW(pszFullPath, dwCbFullPath, L"%s%s%s", pszRootName,
                        ((pszKeyPath[0] == L'\\') ? L"" : L"\\"), pszKeyPath);
    }
    else
    {
        StringCbCopyW(pszFullPath, dwCbFullPath, pszRootName);
    }

    SendMessageW(hStatusBar, SB_SETTEXTW, 0, (LPARAM)pszFullPath);
    free(pszFullPath);
```